src/main/java/io/deepstream/message/Connection.java

```diff
@@ -164,4 +166,10 @@ private Endpoint createEndpoint(String url, Map options) throws Exception {
         }
         return endpoint;
     }
+
+    private String getAuthData( String data ) {
+        String result;
```

Comment:
Guessing login failed should be 'Object' instead of String and force the user to cast it to string if needed. Thisales more sense for now though